### ISSUE TITLE
Use f-strings instead of .format

### ISSUE THREAD

**Issue by charliestrawn** [nats-io/nats.py#105]
Python 3.6+ supports a new style of string formatting which reads much easier.

```py
uri = urlparse("{}://{}".format(scheme, connect_url))
```
becomes
```py
uri = urlparse(f'{scheme}://{connect_url}')
```
This reads nicer in areas like the protocol where we are concatting strings with `+` as well.

**Comment by charliestrawn** [comment]
I'd like to tackle this @wallyqs just wanted to call it out in issues as to get your thoughts and to not duplicate effort.

**Comment by wallyqs** [comment]
@charliestrawn this sounds good to me! I was thinking of cutting a release for 0.9.0 today, would you like me to wait or can we include this in `v0.9.1` maybe?

**Comment by wallyqs** [comment]
Actually had already uploaded `asyncio-nats-client 0.9.0` so we can add this to the next release. Thanks Charlie.

**Comment by charliestrawn** [comment]
Yep no rush, totally a cosmetic update.

**Comment by Bernolt** [comment]
Hello, if you still want to do this change. I could get a jump on it.

**Comment by wallyqs** [comment]
@Bernolt that would be great, thanks!

**Comment by Bernolt** [comment]
Actually, I ran a pyupgrade to get rid of them, is additinally got rid of:
- old-style class definition
```
class C(object): pass     # class C: pass
class C(B, object): pass  # class C(B): pass
```
- super calls
```
class C(Base):
    def f(self):
        super(C, self).f()   # super().f()
```
and of course switched f-strings instead of .format.

Do you mind if I keep those changes or you would prefer to stick to the f-string change only?

_Also, f-string change will break compatibily with version of python below 3.6._ Are you ok with that?

**Comment by wallyqs** [comment]
Thanks for the PR, I think it is fine to allow for 3.6 as the minimum version.

**Comment by charliestrawn** [comment]
Yeah I'm pretty sure we are already using some things that bind us to 3.6 or later. Closing as addressed in #152
